### PR TITLE
Add StepGroup and Step components

### PR DESCRIPTION
**Pull request by lucasecdb**
#### What problem is this solving?
Adds the initial version of the `StepGroup` and `Step` components, and their respective blocks interfaces.

[Related clubhouse story](https://app.clubhouse.io/vtex/story/29567/componente-stepper-visual)

#### How should this be manually tested?
[Workspace](https://chkio--checkoutio.myvtex.com/checkout/#/)

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [x] Linked this PR to a Clubhouse story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Screenshots or example usage

image

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
✔️ | New feature 
_ | Breaking change 
_ | Technical improvements

### CODE REVIEW

**Review by jeffersontuc — APPROVED** [review]

**Comment by marcoskwkm** [comment]
Could you add to the PR description a workspace with an example usage of these components, like the one you showed us yesterday?

**Comment by lucasecdb** [comment]
sure! just a second. I'll use the template I've added

File: react/StepGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+  const cssHandles = useCssHandles(classes)
+
+  useLayoutEffect(() => {
+    if (assigning.current) {
+      // At this point all of the children have pushed into the array so we set
+      // assigning to false and force an update. Since we're in
+      // useLayoutEffect, we won't get a flash of rendered content, it will all
+      // happen synchronously. And now that this is false, children won't push
+      // into the array on the forceUpdate      assigning.current = false
```
Comment (marcoskwkm) [inline react/StepGroup.tsx]: nit: Leftover `assigning.current = false` in comment.

File: react/StepGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+import { useCssHandles } from 'vtex.css-handles'
+
+/**
+ * The component below is implemented following the "guidelines" of
+ * this [1] gist. It is known that creating compound components in React
+ * that have a strong relation of parent-child (like a select and their
+ * option's) is hard, and there is currently no good way to implement this
+ * because of the lack of primitives in the library.
+ *
+ * [1] https://gist.github.com/ryanflorence/10e9387f633f9d2e6f444a9bddaabf6e
```
Comment (marcoskwkm) [inline react/StepGroup.tsx:18]: Nice hack

**Review by darlenedms — APPROVED** [review]

File: react/Step.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      return
+    }
+
+    indexRef.current = items.current.push(elementRef.current!) - 1
+    updateParent()
+  })
+
+  return {
+    // first render its wrong, after a forceUpdate in parent useLayoutEffect it's
+    // right, and its all synchronous so we don't get any flashing
```
Comment (marcoskwkm) [inline react/Step.tsx]: I would rephrase this as something like
`index is wrong on the first render, but the parent forces a synchronous second render that fixes its value without flashing the component`

**Review by marcoskwkm — APPROVED** [review]

File: react/StepGroup.tsx (hunk 1 above)
Comment (lucasecdb) [reply react/StepGroup.tsx]: ohhhh, so there it was..

File: react/Step.tsx (hunk 3 above)
Comment (lucasecdb) [reply react/Step.tsx]: nice
